src/main/java/net/snowflake/ingest/streaming/internal/ChannelRuntimeState.java

```diff
@@ -56,12 +61,21 @@ long getRowSequencer() {
     return rowSequencer.get();
   }
 
+  /** @return the start offset token */
+  String getStartOffsetToken() {
+    return startOffsetToken;
+  }
+
   /**
    * Updates the channel's offset token.
    *
    * @param offsetToken new offset token
    */
   void setOffsetToken(String offsetToken) {
+    if (!startOffsetTokenSet) {
+      startOffsetToken = offsetToken;
```

Comment:
Not sure why we're setting the startOffsetToken to offsetToken, offsetToken is the last offset of the first batch, so basically we're sending (last offset of first batch, last offset of the chunk) which won't be continuous from the previous chunk, also the row count won't be matched.

Reply:
This is where we get offset token for a batch, no? We copy the data here from `ChannelRuntimeState` for a `RowBuffer` into a `ChannelDTO` or something further down the line. So with `startOffsetTokenSet` we essentially grab the FIRST offset token provided in a call to `insertRows` for a batch and store that.

Reply:
If you see how ChannelRuntimeState is being constructed, it's only being constructed once in the SnowflakeStreamingIngestChannelInternal and never being reseted between flushes. You can run a test to verify, but I think this means the startOffsetToken will always be the same for all chunks.

Reply:
I updated the logic in `ChannelRuntimeState` already to store the correct start and end offset, so this is not needed anymore